### PR TITLE
Remove AdditionalInfo (AdditionalAttestationData)

### PR DESCRIPTION
**Pull request by rbehjati**
Ref https://github.com/project-oak/oak/issues/2911

We currently don't have a use for the `AdditionalAttestationData` field in `ServerIdentity`. That part requires more thinking and redesign. This is a cleanup PR to remove the existing unused code.

### CODE REVIEW

File: remote_attestation/rust/src/message.rs [hunk 1]
```diff
@@ -236,25 +227,21 @@ impl ServerIdentity {
             + REPLAY_PROTECTION_ARRAY_LENGTH
             + SIGNATURE_LENGTH
             + SIGNING_ALGORITHM_KEY_LENGTH
-            + 2 * VEC_SIZE_PREFIX_LENGTH
+            + VEC_SIZE_PREFIX_LENGTH
```
Comment (mariaschett) [inline remote_attestation/rust/src/message.rs:230]: Why did we lose the 2* here?

**Review by mariaschett — APPROVED** [review]
Looks great! Thank you!

File: oak_functions/client/rust/README.md [hunk 2]
```diff
@@ -23,13 +23,21 @@ Example invocation:
...
 
-As part of the Remote Attestation handshake the server sends to the client some
-additional info along with the attestation quote. The client verifies this
-additional information according to its policy, before sending any requests to
-the server. If the additional information contains a LogEntry from Rekor and an
-accompanying endorsement file, then the client verifies the LogEntry and the
-content of the endorsement file. In particular, the SHA256 hash of the binary in
-the subject of the endorsement file must be the same as the hash of the binary
-received in the attestation quote. Checking the LogEntry involves verifying the
+The client may have a privacy policy that the server must conform to. If the
```
[the remark this answers is not in the document]
Comment (mariaschett) [reply oak_functions/client/rust/README.md:26]: Follow up thought: maybe we want to move the info into an issue and link the issue here instead? Then it is clear that we still need to design this.

[the remark this answers is not in the document]
Comment (rbehjati) [reply oak_functions/client/rust/README.md:26]: I think creating an issue is a good idea. But I don't have enough context for that at the moment. We should have a discussion about that soon. Also I think this part of the README should be moved one level up to `oak_functions/client/`. I've created #2998. I am not sure if it is a good idea to include a TODO here in the README. Let me know if you think it is a good idea.

File: remote_attestation/java/src/com/google/oak/remote_attestation/ClientHandshaker.java [hunk 3]
```diff
@@ -218,13 +208,8 @@ private Boolean verifyAttestationInfo(Message.ServerIdentity serverIdentity) thr
     return true;
   }
 
-  private boolean verifyAdditionalAttestationData(byte[] additionalAttestationData) {
-    // TODO(#2917): Implement this method.
```
[the remark this answers is not in the document]
Comment (rbehjati) [reply remote_attestation/java/src/com/google/oak/remote_attestation/ClientHandshaker.java:222]: Yes. But this is verification of additional attestation data. We have a separate method `verifyRemoteAttestationReport` which already points to #2917.

File: remote_attestation/java/src/com/google/oak/remote_attestation/ClientHandshaker.java [hunk 4]
```diff
@@ -234,9 +219,8 @@ private byte[] sha256(byte[] value) {
   }
 
   /**
-   * Appends {@code serializedMessage} to the protocol transcript.
-   * Transcript is a concatenation of all sent and received messages, which is used for preventing
-   * replay attacks.
+   * Appends {@code serializedMessage} to the protocol transcript. Transcript is a concatenation of
```
[the remark this answers is not in the document]
Comment (rbehjati) [reply remote_attestation/java/src/com/google/oak/remote_attestation/ClientHandshaker.java]: Fixed, if this is what you mean: 

```suggestion
   * Appends {@code serializedMessage} to the protocol transcript. The protocol transcript is a concatenation of
```

File: remote_attestation/rust/src/message.rs (hunk 1 above)
Comment (rbehjati) [reply remote_attestation/rust/src/message.rs:230]: Because one field with varying length (i.e., additional info) is removed.

**Review by rbehjati — COMMENTED** [review]
Thanks for the quick review :)

File: experimental/tf_proxy/server/src/main.rs [hunk 5]
```diff
@@ -91,21 +89,11 @@ async fn main() -> anyhow::Result<()> {
     let backend = grpc::BackendConnection::connect(SOCKET).await;
     let client = backend.create_client();
 
-    // Create fake configuration report for now, as it cannot be empty for the attestation
-    // handshake.
-    // TODO(#2420): Remove once Java client can work without the configuration report.
```
[the remark this answers is not in the document]
Comment (rbehjati) [reply experimental/tf_proxy/server/src/main.rs:96]: I think we still want to keep it, even thought it is not mentioned in a TODO anywhere in the code. The Java and Rust implementations must be consistent.

File: remote_attestation/rust/src/message.rs (hunk 1 above)
Comment (rbehjati) [reply remote_attestation/rust/src/message.rs]: Added clarifying comment.

**Review by ipetr0v — APPROVED** [review]

File: experimental/tf_proxy/server/src/main.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (rbehjati) [reply experimental/tf_proxy/server/src/main.rs:96]: Added a TODO on `verifyAttestationInfo` in the remote attestation java client.

**Pull request merged**